sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/SubstationServiceTest.java

```diff
@@ -0,0 +1,133 @@
+// SPDX-FileCopyrightText: 2022 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.scl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.lfenergy.compas.scl2007b4.model.SCL;
+import org.lfenergy.compas.scl2007b4.model.TLNode;
+import org.lfenergy.compas.scl2007b4.model.TSubstation;
+import org.lfenergy.compas.sct.commons.exception.ScdException;
+import org.lfenergy.compas.sct.commons.testhelpers.SclTestMarshaller;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class SubstationServiceTest {
+
+    @ParameterizedTest
+    @ValueSource(strings = {"/scd-substation-import-ssd/ssd_with_2_substations.xml", "/scd-substation-import-ssd/ssd_without_substations.xml"})
+    void testAddSubstation_Check_SSD_Validity(String ssdFileName) throws Exception {
```

Comment:
It could be useful to test the exception reason and to rename test accordingly

Reply:
OK